### PR TITLE
Fix for potential warnings

### PR DESCRIPTION
**Pull request by MapleLeaf-X**
A small fix for potential warnings C4390 / -Wempty-body and C4706.

PS: Is this pr actually attempting to merge 3 commits? and if so should I change that?

### CODE REVIEW

File: src/common/uiactioncmn.cpp [hunk 1]
```diff
@@ -203,13 +203,21 @@ bool wxUIActionSimulator::Select(const wxString& text)
 #endif // wxUSE_COMBOBOX
 #if wxUSE_CHOICE
     wxChoice* choice;
-    if ( !container && (choice = wxDynamicCast(focus, wxChoice)) )
-        container = choice;
+    if ( !container )
+    {
+        choice = wxDynamicCast(focus, wxChoice);
+        if ( choice )
+            container = choice;
```
Comment (vadz) [inline src/common/uiactioncmn.cpp:210]: I'd write this as

```suggestion
        if ( auto choice = wxDynamicCast(focus, wxChoice) )
            container = choice;
```

(and remove `wxChoice* choice` declaration above, of course).

**Review by vadz — COMMENTED** [review]
Thanks, I can cherry pick the last commit here, but please make your PR based on wx master and not on your local branch into which master was merged in the future.

Also, please check that `git config user.email` returns something reasonable for you (even if it's just `MapleLeaf-X@users.noreply.github.com`) and set it if necessary. TIA!

File: src/common/uiactioncmn.cpp (hunk 1 above)
Comment (MapleLeaf-X) [reply src/common/uiactioncmn.cpp:210]: git config user.email was actually empty so thanks for informing me, I have it fixed now. as for the code I will make some changes then and sort out this strange commit issue.

**Comment by MapleLeaf-X** [comment]
I'm gonna need to close this for now until I'll resolve some issues on my end

**Pull request closed without merging**